platform/ios/PolyPodApp/PodApi/Endpoint/Endpoint.swift

```diff
@@ -31,80 +31,114 @@ struct EndpointInfo: Decodable {
 }
 
 final class Endpoint: EndpointProtocol {
-    
+
     init() {
         delegate = nil
     }
-    
+
     let network: Network = Network()
     var delegate: EndpointDelegate?
-    
-    func approveEndpointFetch(endpointId: String, completion: @escaping (Bool) -> Void) {
+
+    private func approveEndpointFetch(endpointId: String, completion: @escaping (Bool) -> Void) {
         delegate?.doHandleApproveEndpointFetch(endpointId: endpointId, completion: completion)
     }
-    
-    private func endpointInfoFromId(endpointId: String) -> EndpointInfo? {
-        let endpointsPath = 
-            Bundle.main.bundleURL.appendingPathComponent("config/endpoints.json")
-        guard let endpointsJsonData = (try? Data(contentsOf: endpointsPath)) else { return nil }
-        guard let endpointsJson = (
-            try? JSONDecoder().decode(
-                Dictionary<String, EndpointInfo>.self, 
-                from: endpointsJsonData
-                )
-            ) else { 
-                return nil 
+
+    private func jsonStringify(_ objectToConvert: Any) -> String {
+        var resultJson = ""
+        if let json = try? JSONSerialization.data(withJSONObject: objectToConvert) {
+            if let content = String(data: json, encoding: String.Encoding.utf8) {
+                // here `content` is the JSON dictionary containing the String
+                resultJson = content
             }
-        return endpointsJson[endpointId]
+        }
+        return resultJson
+    }
+
+    func uploadToServer(
+        errorMsg: String,
+        endpointId: String,
+        completionHandler: @escaping (Error?) -> Void
+    ) {
+        guard let endpointInfo = self.endpointInfoFromId(endpointId: endpointId) else {
+            Log.error("uploadToServer failed: No endpoint found for: \(endpointId)")
+            completionHandler(
+                PodApiError.endpointError("post, No endpoint found for: \(endpointId)")
+            )
+            return
+        }
+        let payload = jsonStringify([errorMsg])
+        if payload.isEmpty {
+            completionHandler(PodApiError.endpointError("Empty payload!"))
+        }
+
+        let response = self.network.httpPost(
+            url: endpointInfo.url,
+            body: payload,
+            contentType: "application/json",
+            authToken: endpointInfo.auth,
+            allowInsecure: endpointInfo.allowInsecure
+        )
+
+        switch response {
+        case .failure(let error):
+            Log.error("uploadToServer(): endpoint.post to \(endpointId) failed with: \(error.localizedDescription)")
+            let errorCode = response.map { "failed with code \($0)" }
+            completionHandler(PodApiError.endpointError("post \(errorCode)"))
```

Comment:
Nit: This is inconsistent with Android, but I guess this how `PodApiError.endpointError` was originally designed.